project2/py/transform.py

```diff
@@ -123,22 +132,27 @@ def substitute_ingredients(recipe, predicate, substituter):
     return new_recipe
 
 
-def fix_directions(old_ingredient_name, new_ingredient_name, directions):
+def _fix_directions(old_ingredient_name, new_ingredient_name, directions):
     """
         substitute old ingredient with new ingredient in all directions
     """
     new_directions = utils.pmap(
-        lambda d: fix_direction(
+        lambda d: _fix_direction(
             old_ingredient_name,
             new_ingredient_name,
             d),
         directions)
     return new_directions
 
 
-def fix_direction(old_ingredient_name, new_ingredient_name, direction):
+def _fix_direction(old_ingredient_name, new_ingredient_name, direction):
     """
         substitute old ingredient with new ingredient in direction
     """
     new_direction = direction.replace(old_ingredient_name, new_ingredient_name)
+
+    tokens = old_ingredient_name.split()
+    for token in tokens:
+        new_direction = new_direction.replace(token, new_ingredient_name)
```

Comment:
this addition will find ingredients that are referred to by a shorter name than is used in the ingredients list e.g. 'chicken thighs' will often be referred to as 'chicken' in directions